Commit:     6322555dbeb46c512d6be42c3829087af4e4db9d
Parent:     7afe7b5969329175ac4f55a6b9c13ba4f6dc267e
Refname:    refs/heads/master
Web:        https://git.kernel.org/torvalds/c/6322555dbeb46c512d6be42c3829087af4e4db9d
Author:     Icenowy Zheng <uwu@icenowy.me>
AuthorDate: Fri May 26 18:08:01 2023 +0800
Committer:  Matthias Brugger <matthias.bgg@gmail.com>
CommitDate: Thu Jun 15 13:14:57 2023 +0200

    arm64: dts: mediatek: mt8173-elm: remove panel model number in DT
    
    Currently a specific panel number is used in the Elm DTSI, which is
    corresponded to a 12" panel. However, according to the official Chrome
    OS devices document, Elm refers to Acer Chromebook R13, which, as the
    name specifies, uses a 13.3" panel, which comes with EDID information.
    
    As the kernel currently prioritizes the hardcoded timing parameters
    matched with the panel number compatible, a wrong timing will be applied
    to the 13.3" panel on Acer Chromebook R13, which leads to blank display.
    
    Because the Elm DTSI is shared with Hana board, and Hana corresponds to
    multiple devices from 11" to 14", a certain panel model number shouldn't
    be present, and driving the panel according to its EDID information is
    necessary.
    
    Signed-off-by: Icenowy Zheng <uwu@icenowy.me>
    Reviewed-by: AngeloGioacchino Del Regno <angelogioacchino.delregno@collabora.com>
    Reviewed-by: Douglas Anderson <dianders@chromium.org>
    Link: https://lore.kernel.org/r/20230526100801.16310-1-uwu@icenowy.me
    Signed-off-by: Matthias Brugger <matthias.bgg@gmail.com>
---
 arch/arm64/boot/dts/mediatek/mt8173-elm.dtsi | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/mediatek/mt8173-elm.dtsi b/arch/arm64/boot/dts/mediatek/mt8173-elm.dtsi
index 03d1ab2ca8203..111495622cacd 100644
--- a/arch/arm64/boot/dts/mediatek/mt8173-elm.dtsi
+++ b/arch/arm64/boot/dts/mediatek/mt8173-elm.dtsi
@@ -287,7 +287,7 @@
 
 		aux-bus {
 			panel: panel {
-				compatible = "lg,lp120up1";
+				compatible = "edp-panel";
 				power-supply = <&panel_fixed_3v3>;
 				backlight = <&backlight>;
 